include/envoy/ssl/context_config.h

```diff
@@ -84,6 +84,11 @@ class ContextConfig {
    * @return the set of capabilities for handshaker instances created by this context.
    */
   virtual HandshakerCapabilities capabilities() const PURE;
+
+  /**
+   * @return a callback for configuring an SSL_CTX before use.
+   */
+  virtual SslCtxCb sslctx_cb() const PURE;
```

Comment:
I think this might be simpler to make this `virtual void setupSslCtx(SSL_CTX* ctx) PURE;`

Reply:
This would be simpler, but I like the mechanism discussed in https://github.com/envoyproxy/envoy/pull/15179#discussion_r583964639 where we get to check if the cb exists before calling it, so I can't do this too.